Modules/Scripted/VPAWVisualize/VPAWVisualize.py

```diff
@@ -322,25 +336,135 @@ def onApplyButton(self):
         ):
 
             # Compute output
-            self.logic.process(
-                self.ui.inputSelector.currentNode(),
-                self.ui.outputSelector.currentNode(),
-                self.ui.imageThresholdSliderWidget.value,
-                self.ui.invertOutputCheckBox.checked,
+            list_of_files = self.logic.process(
+                self.ui.DataDirectory.currentPath, self.ui.PatientPrefix.text
             )
-
-            # Compute inverted output (if needed)
-            if self.ui.invertedOutputSelector.currentNode():
-                # If additional output volume is selected then result with inverted
-                # threshold is written there
-                self.logic.process(
-                    self.ui.inputSelector.currentNode(),
-                    self.ui.invertedOutputSelector.currentNode(),
-                    self.ui.imageThresholdSliderWidget.value,
-                    not self.ui.invertOutputCheckBox.checked,
-                    showResult=False,
+            # Display output
+            self.clearOutputWidgets()
+            self.addOutputWidgets(list_of_files)
+
+    def clearOutputWidgets(self):
+        slicer.mrmlScene.GetSubjectHierarchyNode().RemoveAllItems(True)
+        self.show_nodes = list()
+
+    def processP3(self, filename, properties):
+        with open(filename, "rb") as f:
+            contents = pk.load(f)
+
+        print(f"File type for {filename} is not currently supported")
+        print(f"{filename} contains {summary_repr(contents)}") # !!!
+        # !!! Create node from contents
+
+        return None
+
+    def createNode(self, filename, basename_repr, props):
```

Comment:
This could use a docstring

Also, it belongs in `VPAWVisualizeLogic` rather than `VPAWVisualizeWidget`.

In general any functionality that in principle doesn't need a UI to work should go into the Logic rather than the Widget class. The Widget class can call the appropriate logic functions based on UI signals, and the logic class should have all the self-contained data processing, scene manipulation, etc.

Reply:
I was thinking that manipulation of an actual `node` object means to use `VPAWVisualizeWidget`, but I have moved this (and similar functions called by `VPAWVisualizeWidget.onApplyButton`) to `VPAWVisualizeLogic`.